test/saucelabs/swisssearch_test.py

```diff
@@ -1,223 +1,253 @@
 # -*- coding: utf-8 -*-
 
 import time
+import sys
 from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.common.by import By
 from selenium.webdriver.support import expected_conditions as EC
-from helpers import bCheckIfUrlHasChanged
 
-# Swisssearch Test Using saucelabs
+DEFAULT_WAIT = 10
+# Gloabl css selectors
+inputContainerClass = 'ga-search-input-container'
+dropDownClass = 'ga-search-dropdown'
+itemClass = 'ga-search-item'
 
-QUERYSTRING_OF_RARON = "X=128114.80&Y=629758.13&zoom=10"
-QUERYSTRING_OF_RARON_KANALSTRASSE = "X=128141.00&Y=627443.00&zoom=10"
-QUERYSTRING_OF_RTE_BERNE_LAUSANNE = "X=154207.56&Y=539257.38&zoom=10"
-QUERYSTRING_OF_PL_CHATEAU_AVENCHES = "X=192309.80&Y=569734.06&zoom=10"
-QUERYSTRING_OF_PIAZZA_MESOLCINA_BELLINZONA = "X=117500.00&Y=722500.00&zoom=10"
-QUERYSTRING_MOOS = "X=128630.12&Y=627650.38&zoom=10"
-QUERYSTRING_OF_REALTA = "X=181085.41&Y=751355.13&zoom=10"
-DEFAULT_WAIT = 6
 
-
-def wait_url_changed(driver, old_url, timeout=DEFAULT_WAIT):
-    UrlHasChanged = False
+def waitUrlChange(driver, pattern, find=True, timeout=DEFAULT_WAIT):
     t0 = time.time()
-    while not UrlHasChanged:
-        new_url = driver.current_url
-        if old_url != new_url:
-            UrlHasChanged = True
-        t1 = time.time()
-        if t1 - t0 > timeout:
+    newUrl = driver.current_url
```

Comment:
We have many elements on the page that change their url in exactly the same way as the url in the address bar. Why don't we use those elements and profit from selenium internal detection loops and timeouts?